### PR TITLE
[ODS-5628] Redesign API key/secret authentication to eliminate use of Entity Framework query

### PR DESCRIPTION
**Pull request by gmcelhanon**

### CODE REVIEW

**Comment by gmcelhanon** [comment]
The nature of the changes are essentially that authentication-related components that were previously in EdFi.Admin.DataAccess that weren't shared by other projects have been moved in the EdFi.Ods.Api project in the main solution. Additionally, obsolete methods have been removed, and interfaces renamed based on the new scope of their behavior. The high-level flow of an authentication request to the TokenController looks like the following:
image

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/ApiClientAuthenticatorTests.cs [hunk 1]
```diff
@@ -8,147 +8,133 @@
...
+                _apiClientAuthenticator = new ApiClientAuthenticator(apiClientDetailsProvider, secretVerifier);
             }
 
             protected override void Act()
             {
-                _actualResult = _apiClientAuthenticator.TryAuthenticate("MyKey", "MyInvalidSecret", out _apiClientIdentity);
+                _actualResult = _apiClientAuthenticator.TryAuthenticateAsync("MyKey", "MyInvalidSecret")
+                    .ConfigureAwait(false)
+                    .GetAwaiter()
+                    .GetResult();;
```
Comment (ericstockdale) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/ApiClientAuthenticatorTests.cs:125]: Remove redundant semi-colon

Comment (gmcelhanon) [reply Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/ApiClientAuthenticatorTests.cs:125]: Actually found 3 of them in that file... cleaned up locally and will commit and push all other potential changes once you are done with your review.

File: Application/EdFi.Ods.Api/ScheduledJobs/Jobs/DeleteExpiredTokensScheduledJob.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿using System;
```
Comment (ericstockdale) [inline Application/EdFi.Ods.Api/ScheduledJobs/Jobs/DeleteExpiredTokensScheduledJob.cs:1]: Can you add the license header here.  I am pretty sure I did this file originally and clearly forgot it :-(

**Comment by ericstockdale** [comment]
@gmcelhanon Just a few comments (one of which you already addressed)

**Comment by ericstockdale** [comment]
@gmcelhanon Unfortunately it also looks like there are merge conflicts now

File: Application/EdFi.Ods.Api/ScheduledJobs/Jobs/DeleteExpiredTokensScheduledJob.cs (hunk 2 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/ScheduledJobs/Jobs/DeleteExpiredTokensScheduledJob.cs:1]: Done.

**Review by ericstockdale — APPROVED** [review]

**Pull request merged**